marklogic-data-hub/src/test/java/com/marklogic/hub/security/RedactionUserTest.java

```diff
@@ -0,0 +1,85 @@
+package com.marklogic.hub.security;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.appdeployer.command.schemas.LoadSchemasCommand;
+import com.marklogic.client.ResourceNotFoundException;
+import com.marklogic.client.io.DocumentMetadataHandle;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.hub.AbstractHubCoreTest;
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.flow.FlowInputs;
+import com.marklogic.hub.flow.impl.FlowRunnerImpl;
+import com.marklogic.mgmt.resource.databases.DatabaseManager;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static com.marklogic.client.io.DocumentMetadataHandle.Capability.READ;
+import static com.marklogic.client.io.DocumentMetadataHandle.Capability.UPDATE;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class RedactionUserTest extends AbstractHubCoreTest {
+
+    @BeforeEach
+    void setUp() {
+        runAsDataHubDeveloper();
+        installProjectInFolder("test-projects/redaction-test");
+        new LoadSchemasCommand().execute(newCommandContext());
+
+        ingestDocument();
+    }
+
+    @AfterEach
+    void tearDown() {
+        runAsAdmin();
+        new DatabaseManager(getHubConfig().getManageClient()).clearDatabase(HubConfig.DEFAULT_STAGING_SCHEMAS_DB_NAME);
+    }
+
+    @Test
+    void testRedactionAsDataHubOperator() {
+        runFlowAsUser(this::runAsDataHubOperator);
+        verifyRedactedDoc();
+    }
+
+    @Test
+    void testRedactionAsHubCentralOperator() {
+        runFlowAsUser(() -> runAsTestUserWithRoles("hub-central-operator"));
+        verifyRedactedDoc();
+    }
+
+    @Test
+    void testRedactionAsAForbiddenUser() {
+        runFlowAsUser(() -> runAsTestUserWithRoles("hub-central-step-runner"));
+        verifyRedactedDocNotPresent();
```

Comment:
I think we still need to verify that the RunFlowResponse contains an error in stepOutput for the step that failed. It's good to verify that the doc wasn't written too, but the most important thing to verify is that the step failed, and the way to do that is via the RunFlowResponse.

Reply:
Done